extensions/ql-vscode/src/extension.ts

```diff
@@ -1557,6 +1563,53 @@ async function activateWithInstalledDistribution(
   };
 }
 
+function addUnhandledRejectionListener() {
+  const handler = (error: unknown) => {
+    // This listener will be triggered for errors from other extensions as
+    // well as errors from this extension. We don't want to flood the user
+    // with popups about errors from other extensions, and we don't want to
+    // report them in our telemetry.
+    //
+    // We check if an error is from this extension by looking for a specific
+    // path in the stack trace.
+    //
+    // The stack trace gets redacted before being sent as telemetry, but at
+    // this point in the code we have the full unredacted information.
+    const isFromThisExtension = getErrorStack(error).includes(
+      "/extensions/ql-vscode/",
+    );
```

Comment:
Have you tested this using a packaged extension (VSIX)? I believe the stack trace in the packaged extension looks something like shown in [the source map script](https://github.com/github/vscode-codeql/blob/56d283f6d5f9c0ebd4e0e2857847ccf22549a8ad/extensions/ql-vscode/scripts/source-map.ts):

```
Error: Failed to find CodeQL distribution.
     at CodeQLCliServer.getCodeQlPath (/Users/user/.vscode/extensions/github.vscode-codeql-1.7.8/out/extension.js:131164:13)
     at CodeQLCliServer.launchProcess (/Users/user/.vscode/extensions/github.vscode-codeql-1.7.8/out/extension.js:131169:24)
     at CodeQLCliServer.runCodeQlCliInternal (/Users/user/.vscode/extensions/github.vscode-codeql-1.7.8/out/extension.js:131194:24)
     at CodeQLCliServer.runJsonCodeQlCliCommand (/Users/user/.vscode/extensions/github.vscode-codeql-1.7.8/out/extension.js:131330:20)
     at CodeQLCliServer.resolveRam (/Users/user/.vscode/extensions/github.vscode-codeql-1.7.8/out/extension.js:131455:12)
     at QueryServerClient2.startQueryServerImpl (/Users/user/.vscode/extensions/github.vscode-codeql-1.7.8/out/extension.js:138618:21)
```

This does not include `/extensions/ql-vscode` because when the extension is packaged, that part of the path is not included.

Reply:
Good point, I was testing with the development host rather than a packaged extension. If the paths do always look like the ones you've included here, that looks very promising for a robust way to detect extension code.

Reply:
I've rewritten this check to use `extension.extensionPath` instead, which will be much more accurate. It'll also avoid issues on windows because I accidentally used forward slashes in the previous implementation.